metricbeat/mb/testing/testdata.go

```diff
@@ -193,6 +209,13 @@ func TestMetricsetFieldsDocumented(t *testing.T, metricSet mb.MetricSet, events
 }
 
 func runTest(t *testing.T, file string, module, metricSetName string, config DataConfig) {
+	filename := filepath.Base(file)
```

Comment:
Why is it needed now? What was different in the past? Was the `data.json` included in the testing somehow?

Could you add an explanatory comment of what this code block is trying to achieve?

Reply:
> Was the data.json included in the testing somehow?

Yes.

We also were not using the custom write path, although it should be possible before. I'm guessing it was a distraction.

Reply:
> Yes.

How? I'm not about this change and how it affects the end result, could you elaborate more on this on why we exclude this file and what was the behavior before?

Then it would be nice to have this as a comment in the code please so as to know what these lines intend to do.

Reply:
Yes, I wrote this comment to explain it:
```
If the expected suffix is '.json' we need to exclude the data.json file, since by the end of this function we may create a data.json file if there is a docs file with the config suffix:
			if strings.HasSuffix(file, "docs."+config.Suffix) {
				writeDataJSON(t, data[0], filepath.Join(config.WritePath, "data.json"))
			}
Since the expected file name is obtained through filename + expectedExtension, we could end up testing files like:
			Metrics file: data.json
			Expected metrics file: data.json-expected.json
If the config extension is '.json'.
This is not possible, since running go -test json does not produce an expected file for data.json files. This is why we need to exclude this file from the tests.
```
This is a mistake that wasn't detected before, because `writePath` was never being used.

Reply:
Ok since it fixes an existent issue.